clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1292,8 +1292,10 @@ public void handleResponse(AbstractResponse response) {
                 reenqueue();
             } else if (error == Errors.COORDINATOR_LOAD_IN_PROGRESS || error == Errors.CONCURRENT_TRANSACTIONS) {
                 reenqueue();
-            } else if (error == Errors.TRANSACTIONAL_ID_AUTHORIZATION_FAILED ||
-                    error == Errors.CLUSTER_AUTHORIZATION_FAILED) {
+            } else if (error == Errors.CLUSTER_AUTHORIZATION_FAILED) {
+                log.warn("Retrying upon missing initProducerId due to cluster authorization error");
+                reenqueue();
```

Comment:
Hmm, I think we do want the error to get propagated back to the application. Most users would expect the application to fail so that they have a chance to see the issue and fix it. High level, what I was thinking we could do is fail the request by calling `result.fail`, then transition back to `UNINITIALIZED`. Then if the user wants to, they can call `initTransactions()` to retry.